src/Shared/ProcessExtensions.cs

```diff
@@ -14,100 +11,31 @@ internal static class ProcessExtensions
     {
         public static void KillTree(this Process process, int timeoutMilliseconds)
         {
+#if NETCOREAPP
+            process.Kill(entireProcessTree: true);
```

Comment:
I believe this first came into existence for .NET Core 3.0. This is very likely irrelevant, but I heard there's one customer left on extended support for .NET Core 2.1, and this would break them; can we make the ifdef a little more specific?

Reply:
I believe they're windows-only, so the "throw new System.NotSupportedException" shouldn't affect them.

Reply:
Personally, I prefer to clean up any code that is for versions that EOLed some time ago.

Should I change to use `NETCOREAPP3_0_OR_GREATER` and include the previous implementation (which got replaced by throwing `NotSupportedException`)?

Reply:
> Say it's a faithful replication except it is ok with taking an SDK newer than the runtime. It could grab this MSBuild then load its own runtime and crash.

It'll already do that. We target .NET 7 and some unchanged code will use functionality that is not present on older runtimes.

> You can say that isn't a supported scenario

I can and do say that.



> Should I change to use `NETCOREAPP3_0_OR_GREATER` and include the previous implementation (which got replaced by throwing `NotSupportedException`)?

No thank you, I think it's great as-is.